### PR TITLE
Upload: fix a TypeError when upload a processed file from the `beforeUpload` hook

### PR DESCRIPTION
**Pull request by fengyuanchen**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by Leopoldthecoder** [comment]
Please write a demo of the bug you're trying to fix.

**Comment by fengyuanchen** [comment]
Here is a demo: https://jsfiddle.net/cfy/53jf2oms/1/

Open the console and choose an image to upload, and then you will see a TypeError in the console.

**Comment by fengyuanchen** [comment]
In line 56 of **ajax.js**, I also add the file name to the form data, just because if miss it, the server side will get a file with the name of `'blob'` (without extension) by default after upload the compressed image file, but we may want a valid name as`'example.jpg'`.

**Comment by fengyuanchen** [comment]
In line 84 of **upload.vue**, the second argument is not used, so I remove it.

File: packages/upload/src/upload.vue [hunk 1]
```diff
@@ -92,6 +92,10 @@ export default {
       if (before && before.then) {
         before.then(processedFile => {
           const fileType = Object.prototype.toString.call(processedFile);
+
+          processedFile.name = rawFile.name;
```
Comment (Leopoldthecoder) [inline packages/upload/src/upload.vue]: Is this line necessary? Because here `processedFile` already has a `name` prop.

Comment (fengyuanchen) [reply packages/upload/src/upload.vue]: A [blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob) object only has two properties: `size` and `type`. In the given demo, the `name` property is assigned by the image compressor.

Comment (Leopoldthecoder) [reply packages/upload/src/upload.vue]: OK

Comment (qusiba) [reply packages/upload/src/upload.vue]: This change will cause an error if here the processedFile is still a File object.

**Pull request merged**